hivemind/client/averaging/allreduce.py

```diff
@@ -30,6 +30,7 @@ def __init__(self, *, group_id: GroupID, tensors: Sequence[torch.Tensor], endpoi
         assert endpoint in ordered_group_endpoints, "endpoint is not a part of the group"
         self.group_id, self.endpoint = group_id, endpoint
```

Comment:
What if we specify client-only participants in separate parameter `client_mode_endpoints` hence we have no need in `part_sizes[i]==0`?

Reply:
Yep, that would make sense. But let us not demolish the test for part_sizes[i]==0 - it is an edge case  that may one day help us find a bigger bug.